yb-voyager/src/tgtdb/yugabytedb.go

```diff
@@ -585,23 +642,50 @@ func (yb *TargetYugabyteDB) IsNonRetryableCopyError(err error) bool {
 	return err != nil && utils.InsensitiveSliceContains(NonRetryCopyErrors, err.Error())
 }
 
-func (yb *TargetYugabyteDB) ExecuteBatch(batch []*Event) error {
-	var err error
-	for i := 0; i < len(batch); i++ {
-		event := batch[i]
-		stmt := event.GetSQLStmt(yb.tconf.Schema)
-		log.Debug(stmt)
-		err = yb.connPool.WithConn(func(conn *pgx.Conn) (bool, error) {
-			_, err := conn.Exec(context.Background(), stmt)
+func (yb *TargetYugabyteDB) ExecuteBatch(migrationUUID uuid.UUID, batch EventBatch) error {
+	err := yb.connPool.WithConn(func(conn *pgx.Conn) (retry bool, err error) {
+		ctx := context.Background()
+		tx, err := conn.BeginTx(ctx, pgx.TxOptions{})
```

Comment:
Yes, we will have to measure. But irrespective, we would need a transaction because we need to transactionally update the state on targetDB. 

But, yes, we should measure the perf again once everything is done.